documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -261,4 +263,302 @@ Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rul
+
+**Why the rule is important**: N/A.
+
+**How to fix the violation**: N/A.
+
+**Impact on generated code**: N/A.
+
+**Examples**: N/A
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R2059" />R2059 UniqueResourcePaths
+**Output Message**: Multiple resource providers are not allowed in a single spec. More than one the resource paths were found: '{0}'.
+
+**Description**: Verifies whether more than one resource providers exists in the specification or not.
+
+**Why the rule is important**: Per the ARM guidelines, each swagger specification must contain one resource provider.
+
+**How to fix the violation**: N/A.
+
+**Impact on generated code**: N/A.
+
+**Examples**: N/A
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R2059" />R2059 UniqueResourcePaths
+**Output Message**: Multiple resource providers are not allowed in a single spec. More than one the resource paths were found: '{0}'.
+
+**Description**: Verifies whether more than one resource providers exists in the specification or not.
+
+**Why the rule is important**: Per the ARM guidelines, each swagger specification must contain one resource provider.
+
+**How to fix the violation**: N/A.
+
+**Impact on generated code**: N/A.
+
+**Examples**: N/A
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R2022" />R2022 XmsExamplesRequired
+**Output Message**: Please provide x-ms-examples describing minimum/maximum property set for response/request payloads for operations.{0}.
+
+**Description**: Verifies whether [x-ms-examples](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/x-ms-examples.md#why-x-ms-examples) are provided for each operation or not.
+
+**Why the rule is important**: [x-ms-examples](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/x-ms-examples.md#why-x-ms-examples) are used in model validation. [Benefits](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/x-ms-examples.md#benefits-of-x-ms-examples-extension)
+
+**How to fix the violation**: Please refer the documentation of [x-ms-examples](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/x-ms-examples.md#why-x-ms-examples).
+
+**Impact on generated code**: N/A.
+
+**Examples**: N/A
```

Comment:
let's link to an example in the "x-ms-examples" documentation, similar to above?